seed/lib/superperms/orgs/permissions.py

```diff
@@ -0,0 +1,166 @@
+         of, or first returned organization from user's orgs.
+    """
+    if user.default_organization:
+        return user.default_organization
+    else:
+        orgs = user.orgs.all()
+        parent_orgs = [org for org in orgs if org.child_orgs.all()]
+        org = orgs[0]
+        if parent_orgs:
+            org = parent_orgs[0]
+        return org
+
+
+class SEEDOrgPermissions(BasePermission):
+    """
+    Control API permissions based on OrganizationUser and HTTP Method
+    i.e. GET/POST etc.
+
+    Any method that can change data (POST, PUT, PATCH,  DELETE) requires
+    ROLE_MEMBER, the rest (GET, OPTIONS, HEAD) require ROLE_VIEWER.
+    All require an authenticated user.
+
+    Override perm_map (and subclass) to change these.
+
+    Requires a queryset attribute on the API View(Set). Since we typically
+    need to override get_queryset to ensure filtering by organization[#f1],
+    you should typically  set to MyModel.objects.none().
+
+    ..[#f1] You can use the OrgQuerySetMixin from seed.utils.api on your
+    view(set)to do this.
+
+    """
+    authenticated_users_only = True
+    perm_map = {
+        'GET': ROLE_VIEWER,
+        'OPTIONS': ROLE_VIEWER,
+        'HEAD': ROLE_VIEWER,
+        'POST': ROLE_MEMBER,
+        'PUT': ROLE_MEMBER,
+        'PATCH': ROLE_MEMBER,
+        'DELETE': ROLE_MEMBER,
+    }
+
+    def has_perm(self, request):
+        """Determine if user has required permission"""
+        # pylint: disable=no-member
+        has_perm = False
+        # defaults to OWNER if not specified.
+        required_perm = self.perm_map.get(request.method, ROLE_OWNER)
+        org_id = get_org_id(request)
+        if not org_id:
+            org = get_user_org(request.user)
+            org_id = getattr(org, 'pk')
+        try:
+            org_user = OrganizationUser.objects.get(
+                user=request.user, organization__id=org_id
+            )
+            has_perm = org_user.role_level >= required_perm
+        except OrganizationUser.DoesNotExist:
+            pass
+        return has_perm
+
+    def has_permission(self, request, view):
+        """Determines if user has correct permissions, called by DRF."""
+        # Workaround to ensure this is not applied
+        # to the root view when using DefaultRouter.
+        if hasattr(view, 'get_queryset'):
+            queryset = view.get_queryset()
+        else:
+            queryset = getattr(view, 'queryset', None)
+
+        assert queryset is not None, (
+            'Cannot apply {} on a view that does not set `.queryset`'
+            ' or have a `.get_queryset()` method.'.format(view.__class__)
+        )
+        return (
+            request.user and
+            (
+                is_authenticated(request.user)
+                or not self.authenticated_users_only
+            )
+            and self.has_perm(request)
+        )
+
+
+class SEEDPublicPermissions(SEEDOrgPermissions):
```

Comment:
To the best of my recollection and comparisons against develop, this is new code.  However, you may be thinking of the superperms/orgs/decorators which should be carrying the same or similar functionality just, obviously, as decorators rather than class level permissions.